fhir/client.go

```diff
@@ -85,22 +101,48 @@ func (c *Client) NewSearchTypeRequest(resourceType string) (*http.Request, error
 // SearchTypeURL generates the URL for a search-type interaction of the given
 // resource type
 func (c *Client) SearchTypeURL(resourceType string) string {
-	return c.Base + "/" + resourceType
+	rel := &url.URL{Path: path.Join(c.baseURL.Path, resourceType)}
+	return c.baseURL.ResolveReference(rel).String()
+}
+
+// NewResourceRequestWithSearch creates a new resource interaction request with an
+// additional FHIR search query and sets JSON Accept header. Otherwise it's
+// identical to http.NewRequest.
+func (c *Client) NewResourceRequestWithSearch(resourceType string, searchQuery url.Values) (*http.Request, error) {
+	rel := &url.URL{Path: path.Join(c.baseURL.Path, resourceType), RawQuery: searchQuery.Encode()}
```

Comment:
Also, I think relative URL's should resolve without the need for joining here. Should somehow use SearchTypeURL here.